### ISSUE TITLE
[libc] fix includes of src/__support/macros/config.h

### ISSUE THREAD

**Issue by nickdesaulniers** [llvm/llvm-project#86579]
A few of these are being cleaned up in #86554 so it's probably best for that to land first. But, I can see latent includes of src/__support/macros/config.h that look wrong, such as the one in libc/src/string/memory_utils/aarch64/inline_memcpy.h which has a comment that it includes that header for the definition of LIBC_INLINE.

LIBC_INLINE is defined in libc/src/__support/macros/attributes.h. As such libc/src/string/memory_utils/aarch64/inline_memcpy.h should include libc/src/__support/macros/attributes.h, not src/__support/macros/config.h and the corresponding cmake+bazel build rules fixed.

Probably there are other places too where a similar pattern is occurring and show be fixed up, too.

I suspect this is a fixup for 29f8e076bd427.

**Comment by divyaxdv** [comment]
@nickdesaulniers I would like to work on this issue!

**Comment by nickdesaulniers** [comment]
Thanks @divyaxdv , all yours. You may want to subscribe to #86554 and watch/wait for that to land first.

**Comment by divyaxdv** [comment]
Hello @nickdesaulniers I tried looking into this and seems like you did few cleanup. Do we need more similar cleanups. I am not sure because I am facing issues while running the bazel test. Initially I tried running the same commands

`cd utils/bazel/llvm-project-overlay`

`bazel query @llvm-project//libc/... | xargs bazel test --config=generic_clang --test_output=errors --test_tag_filters=-nobuildkite --build_tag_filters=-nobuildkite --@llvm-project//libc:mpfr=system`

but I am getting this error while running this command https://pastebin.com/NHQA677f.

Then I tried running the bazel test using the command `bazel test --config=generic_clang @llvm-project//libc/... ` from the bazel directory instead of the llvm-project-overlay directory but still facing issues which I have asked in the libc channel discord. It will be helpful if you can provide some direction. Here is the link to the question asked in the discord.

https://discord.com/channels/636084430946959380/636732994891284500/1224100594244124742

**Comment by nickdesaulniers** [comment]
From discord, it sounds like you're on macos? If so, something has likely bitrot in the bazel build for ensuring we don't build linux files on non-linux hosts.

**Comment by nickdesaulniers** [comment]
> Do we need more similar cleanups.

Yes, these are still necessary.

**Comment by divyaxdv** [comment]
I am getting this error when running test using the `ninja check-libc` 
https://pastebin.com/rTWfvN94
I am new to the llvm. if you can provide some direction for solving this issue then it will helpful.

**Comment by nickdesaulniers** [comment]
Yeah, the issue is that the build has bitrot on Darwin (guessing you're on an aarch64 based Mac?).  I broke this recently in https://github.com/llvm/llvm-project/pull/85955 when I deleted libc/src/__support/OSUtil/darwin/quick_exit.h rather than convert it to libc/src/__support/OSUtil/darwin/quick_exit.cpp (as was done for libc/src/__support/OSUtil/baremetal/quick_exit.cpp as an example).

Currently, all first party developers are developing on Linux hosts; we could use assistance in getting the MacOS build back up to snuff.  If you'd like to help with that, it may be a better place to start than this bug.  We can discuss the details more in a new (not-yet-created) issue, or on [discord](https://discord.gg/xS7Z362).

**Comment by krishna2803** [comment]
Hi! @nickdesaulniers if @divyaxdv isn’t currently working on this and it’s still unresolved, would it be possible for me to take it on? Thanks!

**Comment by nickdesaulniers** [comment]
Sure! Reassigned
